rllib/core/examples/simple_ppo_rl_module.py

```diff
@@ -118,45 +126,44 @@ def __init__(self, config: FCConfig):
         super().__init__()
         self.input_dim = config.input_dim
         self.hidden_layers = config.hidden_layers
-        self.activation = config.activation
-        self.output_activation = config.output_activation
 
+        activation_class = getattr(nn, config.activation, lambda: None)()
         self.layers = []
         self.layers.append(nn.Linear(self.input_dim, self.hidden_layers[0]))
         for i in range(len(self.hidden_layers) - 1):
+            if config.activation != "linear":
+                self.layers.append(activation_class)
             self.layers.append(
                 nn.Linear(self.hidden_layers[i], self.hidden_layers[i + 1])
             )
+
         if config.output_dim is not None:
+            if config.activation != "linear":
+                self.layers.append(activation_class)
             self.layers.append(nn.Linear(self.hidden_layers[-1], config.output_dim))
 
         if config.output_dim is None:
             self.output_dim = config.hidden_layers[-1]
         else:
             self.output_dim = config.output_dim
 
-        self.layers = nn.ModuleList(self.layers)
-        self.activation = getattr(nn, self.activation)()
-        if self.output_activation is not None:
-            self.output_activation = getattr(nn, self.output_activation)()
+        self.layers = nn.Sequential(*self.layers)
+
+        self._input_specs = self.input_specs()
```

Comment:
dumb question: Did this API change? Before I had to pass a method name to the `@check_specs` decorator, now a property?

Reply:
yep, There were some issues with that design (Specifically when you want this to work for sub-classes from a super-class that implements those methods). Making it a property fixes those problems.